Sourcecode/include/mx/api/KeyData.h

```diff
@@ -49,6 +53,14 @@ namespace mx
             // within the part
             int staffIndex;
             
+            // If this is true, the KeyData is in "Non Traditional" mode.
+            // The the 'fifths' and 'mode' member variables will be ignored, and
+            // key data will be read from nonTraditionalPitchData.
+            bool nonTraditionalKey;
+            // The non-tradional accidentals.
+            // Fill this with your custom accidentals and alters.
+            std::vector<PitchData> nonTraditionalPitchData;
```

Comment:
Maybe create a new "Data" struct. "NonTraditionalData"?

Reply:
I had started, or maybe completed this here https://github.com/webern/mx/master/develop...feature/custom-key-signatures

As a reference to how I was thinking about it. Or you could check it out, kick the tires. The main difference is...

> Maybe create a new "Data" struct. "NonTraditionalData"?

Yes, I think that would be better than using PitchData.
I'm sorry that the structure of test files is hard to understand (I don't remember how/why it got that way). You can see in my branch that simply added a test file.

I think it's probably best to add a test file that sort of matches the name of the struct in the event that a better one doesn't seem to already exist.

A lot of the really funky names you see in the tests are related to specific test files.

Some xsd for reference:
```xml
	<xs:complexType name="key">
		<xs:annotation>
			<xs:documentation>The key type represents a key signature. Both traditional and non-traditional key signatures are supported. The optional number attribute refers to staff numbers. If absent, the key signature applies to all staves in the part. Key signatures appear at the start of each system unless the print-object attribute has been set to "no".</xs:documentation>
		</xs:annotation>
		<xs:sequence>
			<xs:choice>
				<xs:group ref="traditional-key"/>
				<xs:group ref="non-traditional-key" minOccurs="0" maxOccurs="unbounded"/>
			</xs:choice>
			<xs:element name="key-octave" type="key-octave" minOccurs="0" maxOccurs="unbounded">
				<xs:annotation>
					<xs:documentation>The optional list of key-octave elements is used to specify in which octave each element of the key signature appears.</xs:documentation>
				</xs:annotation>
			</xs:element>
		</xs:sequence>
		<xs:attribute name="number" type="staff-number"/>
		<xs:attributeGroup ref="print-style"/>
		<xs:attributeGroup ref="print-object"/>
	</xs:complexType>
```

```xml
	<xs:group name="non-traditional-key">
		<xs:annotation>
			<xs:documentation>The non-traditional-key group represents a single alteration within a non-traditional key signature. A sequence of these groups makes up a non-traditional key signature</xs:documentation>
		</xs:annotation>
		<xs:sequence>
			<xs:element name="key-step" type="step">
				<xs:annotation>
					<xs:documentation>Non-traditional key signatures can be represented using the Humdrum/Scot concept of a list of altered tones. The key-step element indicates the pitch step to be altered, represented using the same names as in the step element.</xs:documentation>
				</xs:annotation>
			</xs:element>
			<xs:element name="key-alter" type="semitones">
				<xs:annotation>
					<xs:documentation>Non-traditional key signatures can be represented using the Humdrum/Scot concept of a list of altered tones. The key-alter element represents the alteration for a given pitch step, represented with semitones in the same manner as the alter element.</xs:documentation>
				</xs:annotation>
			</xs:element>
			<xs:element name="key-accidental" type="accidental-value" minOccurs="0">
				<xs:annotation>
					<xs:documentation>Non-traditional key signatures can be represented using the Humdrum/Scot concept of a list of altered tones. The key-accidental element indicates the accidental to be displayed in the key signature, represented in the same manner as the accidental element. It is used for disambiguating microtonal accidentals.</xs:documentation>
				</xs:annotation>
			</xs:element>
		</xs:sequence>
	</xs:group>
```